lib/rouge/lexers/gobstones.rb

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Gobstones < RegexLexer
+      title 'Gobstones'
+      desc 'Gobstones language'
+      tag 'gobstones'
+      filenames '*.gbs'
+
+      def self.analyze_text(_text)
+        0.3
+      end
+
+      reserved = %w(program interactive is return record
+                    field variant case if then else switch to repeat while foreach
+                    in match)
+
+      atoms = %w(False True Verde Rojo Azul Negro Norte Sur Este Oeste)
+
+      types = Set.new
```

Comment:
I'd like to avoid this sort of statefulness if at all possible. Is there any other way to decide whether an id is a type (e.g. beginning with a capital letter)? If not I'd prefer to skip this detection at all.